From acdf673ee14bdb44493855c53d80d0d0f3a1b48c Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Tue, 12 Nov 2019 16:41:21 +0000
Subject: [PATCH] drm/v3d: Plug dma_fence leak

The irq_fence and done_fence are given a reference that is never
released. The necessary dma_fence_put()s seem to have been
deleted in error in an earlier commit.

Fixes: 0b73676836b2 ("drm/v3d: Clock V3D down when not in use.")

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 drivers/gpu/drm/v3d/v3d_gem.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/gpu/drm/v3d/v3d_gem.c
+++ b/drivers/gpu/drm/v3d/v3d_gem.c
@@ -411,6 +411,9 @@ v3d_job_free(struct kref *ref)
 	}
 	kvfree(job->bo);
 
+	dma_fence_put(job->irq_fence);
+	dma_fence_put(job->done_fence);
+
 	v3d_clock_up_put(v3d);
 
 	if (job->perfmon)
